modules/ssl/ssl_engine_init.c

```diff
@@ -189,6 +197,122 @@ static void ssl_add_version_components(apr_pool_t *ptemp, apr_pool_t *pconf,
                  modver, AP_SERVER_BASEVERSION, incver);
 }
 
+#ifdef HAVE_OPENSSL_ECH
+/* 
+ * load any key files we find in the ECHKeyDir directory 
+ * where there are matching <name>.pub and <name>.priv files
+ * that match 
+ */
+static int load_echkeys(SSL_CTX *ctx, const char *echdir, server_rec *s, apr_pool_t *ptemp)
+{
+    /*
+     * Try load any good looking public/private ECH values found in files in that directory
+     *
+     * This code is derived from what I added to openssl s_server, which you can find
+     * in apps/s_server.c in my openssl fork, https://github.com/sftcd/openssl
+     */
+    if (echdir==NULL) {
+        ap_log_error(APLOG_MARK, APLOG_EMERG, 0, s, APLOGNO(10520)
+                "load_echkeys: no directory name - exiting");
+        return -1;
+    }
+    size_t elen=strlen(echdir);
```

Comment:
We still try to keep some c89 compatibility for older compilers on more exotic platforms. Hence please move the declaration of `elen` at the top of the function.

Reply:
sorry, doing that kind of tidying up now generally